src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1422,10 +1426,151 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
+                    constantValue,
+                    isImplicit: true);
+            }
+
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)_compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = argumentNullExceptionMethodSymbol?.ReturnType;
+            var paramNameLiteral = new LiteralOperation(semanticModel, syntax, _compilation.GetSpecialType(SpecialType.System_String), parameter.Name, isImplicit: true);
+            // Occurs when a member is missing.
+            IOperation argumentNullExceptionObject;
+            if (argumentNullExceptionMethodSymbol is null)
+            {
+                argumentNullExceptionObject = new InvalidOperation(
+                    children: ImmutableArray.Create((IOperation)paramNameLiteral),
+                    semanticModel,
+                    syntax,
+                    argumentNullExceptionType,
+                    constantValue,
+                    isImplicit: true);
+            }
+            else
+            {
+                argumentNullExceptionObject = new ObjectCreationOperation(
+                        argumentNullExceptionMethodSymbol,
+                        initializer: null,
+                        ImmutableArray.Create<IArgumentOperation>(
+                            new ArgumentOperation(
+                                paramNameLiteral,
+                                ArgumentKind.Explicit,
+                                parameter: argumentNullExceptionMethodSymbol.Parameters[0],
+                                inConversionOpt: null,
+                                outConversionOpt: null,
+                                semanticModel,
+                                syntax,
+                                isImplicit: true)),
+                        semanticModel,
+                        syntax,
+                        argumentNullExceptionType,
+                        constantValue,
+                        isImplicit: true);
+            }
+
+            IOperation whenTrue = new ExpressionStatementOperation(
+                new ThrowOperation(
+                    argumentNullExceptionObject,
+                    semanticModel,
+                    syntax,
+                    argumentNullExceptionType,
+                    constantValue,
+                    isImplicit: true),
+                semanticModel,
+                syntax,
+                _compilation.GetSpecialType(SpecialType.System_Void),
+                constantValue,
+                isImplicit: true);
+            return new ConditionalOperation(conditionOp, whenTrue, whenFalse: null, isRef: false, semanticModel, syntax, boolType, constantValue, isImplicit: true);
+        }
+
+        private ImmutableArray<ConditionalOperation> GenerateNullChecksForParameters(ImmutableArray<IParameterSymbol> parameters, SyntaxNode syntax, SemanticModel semanticModel)
```

Comment:
>GenerateNullChecksForParameters [](start = 53, length = 31)

This method is only used in one place. Consider inlining this method there, and remove the need for an `ArrayBuilder` or `ImmutableArray`.